graphstorm-processing/.pre-commit-config.yaml

```diff
@@ -0,0 +1,6 @@
+repos:
```

Comment:
What is this?

Reply:
We use [black](https://black.readthedocs.io/en/stable/) to auto-format the code in the project. This is an optional [pre-commit](https://pre-commit.com/) hook that devs can install that ensures `black` is ran on files before they are committed. 

One thing I should do here is limit its file impact to only files under graphstorm-processing.

Reply:
is pre-commit similar to lint? Lint can also do a lot of static checks.